### PR TITLE
Add org.texstudio.TeXstudio.Extension.ngrams.en

### PR DESCRIPTION
**Pull request by picsel2**
This adds an extension of org.texstudio.TeXstudio using the extension point defined in https://github.com/flathub/org.texstudio.TeXstudio/pull/160 .

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* There are none.

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by picsel2** [comment]
I am not sure if the remaining unchecked criteria are actually suitable for extensions of apps that already exist on flathub.

**Comment by hfiguiere** [comment]
It needs https://github.com/flathub/org.texstudio.TeXstudio/pull/160 to land first

**Comment by hfiguiere** [comment]
extensions still need an appstream file

**Comment by Alexander-Wilms** [comment]
The PR adding the extension point has been merged.

**Comment by hfiguiere** [comment]
bot, build org.texstudio.TeXstudio.Extension.ngrams.en

File: org.texstudio.TeXstudio.Extension.ngrams.en.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+id: org.texstudio.TeXstudio.Extension.ngrams.en
+build-extension: true
+runtime: org.texstudio.TeXstudio
```
Comment (hfiguiere) [inline org.texstudio.TeXstudio.Extension.ngrams.en.yaml:3]: you need `runtime-version: stable`

Comment (picsel2) [reply org.texstudio.TeXstudio.Extension.ngrams.en.yaml:3]: Addressed in ea6947e

**Comment by picsel2** [comment]
I don't understand the source of this error messages I get when building now:
```
Cleaning up
Renaming org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml to share/appdata/org.texstudio.TeXstudio.Extension.ngrams.en.appdata.xml
Running appstream-compose
Processing application org.texstudio.TeXstudio.Extension.ngrams.en
Error loading AppData file: no file found for org.texstudio.TeXstudio.Extension.ngrams.en
Error: ERROR: appstream-compose failed: Der Kindprozess wurde mit Status 1 beendet
```

**Comment by Alexander-Wilms** [comment]
Can you give a command to test whether this works correctly and LanguageTool finds the ngram data?

**Comment by picsel2** [comment]
Sure!

1. `flatpak run --command=ls org.texstudio.TeXstudio -la /app/ngrams` should show the language abbreviations of installed extensions as directories ("en" for this PR).
2. Within TeXstudio you can enter the sentence "Don’t forget to put on the breaks.", as described by https://dev.languagetool.org/finding-errors-using-n-gram-data.html . TeXstudio should mark the word "breaks":
   Bildschirmfoto vom 2023-12-08 13-09-00

When building the extension locally I had to pass the argument `--default-branch=stable` to flatpak-builder. Otherwise the extension would have been installed as the branch `master`. The TeXstudio flatpak pulled from flathub from the `stable` branch did not mount the extensions without setting the default branch.

**Comment by Alexander-Wilms** [comment]
I encounter this issue when trying to build it:

```
flatpak-builder --ccache --force-clean --repo=ngrams-repo ngrams org.texstudio.TeXstudio.Extension.ngrams.en.yaml 
Emptying app dir 'ngrams'
Downloading sources
Starting build of org.texstudio.TeXstudio.Extension.ngrams.en
Cache hit for download, skipping build
Cache hit for metainfo, skipping build
Cache miss, checking out last cache hit
Cleaning up
Renaming org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml to share/appdata/org.texstudio.TeXstudio.Extension.ngrams.en.appdata.xml
Running appstream-compose
Processing application org.texstudio.TeXstudio.Extension.ngrams.en
Error loading AppData file: no file found for org.texstudio.TeXstudio.Extension.ngrams.en
Error: ERROR: appstream-compose failed: Child process exited with code 1
```

I tried using the same commands as other extensions (e.g. https://github.com/flathub/org.winehq.Wine.DLLs.dxvk/blob/branch/stable-22.08/org.winehq.Wine.DLLs.dxvk.yml#L86), but that the build still doesn't succeed:
```diff
diff --git a/org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml b/org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml
index a084a47..3acc33d 100644
--- a/org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml
+++ b/org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="addon">
   <id>org.texstudio.TeXstudio.Extension.ngrams.en</id>
+  <extends>org.texstudio.TeXstudio</extends>
   
   <name>English n-gram data</name>
   <summary>English n-gram data for LanguageTool</summary>
@@ -8,10 +9,6 @@
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>CC-BY-3.0</project_license>
   
-  <extends>
-    <id>org.texstudio.TeXstudio</id>
-  </extends>
-  
   <description>
     <p>
       Provides English n-gram data for the LanguageTool server.
diff --git a/org.texstudio.TeXstudio.Extension.ngrams.en.yaml b/org.texstudio.TeXstudio.Extension.ngrams.en.yaml
index dd9411a..6cf6514 100644
--- a/org.texstudio.TeXstudio.Extension.ngrams.en.yaml
+++ b/org.texstudio.TeXstudio.Extension.ngrams.en.yaml
@@ -18,10 +18,13 @@ modules:
           url: https://languagetool.org/download/ngram-data/
           version-pattern: ngrams-en-(\d{8}).zip
           url-template: ngrams-en-$version.zip
+
   - name: metainfo
     buildsystem: simple
     build-commands:
-      - install -Dm644 ${FLATPAK_ID}.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo/
+      - mkdir -p /app/share/metainfo
+      - install -Dm644 org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml ${FLATPAK_DEST}/share/metainfo
+      - appstream-compose --basename ${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}
     sources:
       - type: file
         path: org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml
```

```
flatpak-builder --ccache --force-clean --repo=ngrams-repo ngrams org.texstudio.TeXstudio.Extension.ngrams.en.yaml 
Emptying app dir 'ngrams'
Downloading sources
Starting build of org.texstudio.TeXstudio.Extension.ngrams.en
Cache hit for download, skipping build
Cache miss, checking out last cache hit
========================================================================
Building module metainfo in /home/alexander/Technik/Development/flathub/flathub/.flatpak-builder/build/metainfo-11
========================================================================
Running: mkdir -p /app/share/metainfo
Running: install -Dm644 org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml ${FLATPAK_DEST}/share/metainfo
Running: appstream-compose --basename ${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}
Processing application org.texstudio.TeXstudio.Extension.ngrams.en
Error loading AppData file: no file found for org.texstudio.TeXstudio.Extension.ngrams.en
Error: module metainfo: Child process exited with code 1
```

**Comment by bbhtt** [comment]
a) add `appstream-compose: false` in the top

b) `install -Dm644 org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml ${FLATPAK_DEST}/share/metainfo` is wrong, it is missing a `-t` switch. Should be 

```
install -Dm644 org.texstudio.TeXstudio.Extension.ngrams.en.metainfo.xml -t ${FLATPAK_DEST}/share/metainfo
```

This is the source of the error.

**Comment by barthalion** [comment]
How unreliable would it be using extra-data? I'm somewhat concerned by repackaging something that needs 8GB (plus another 8 for DE?) that will be used by a single app. Is it a widely requested feature?

**Comment by picsel2** [comment]
It looks like I am the single user interested to use this feature. 👀

I understand that adding this amount of data for a single user might be undesirable. Unfortunately using extra-data with such big files has not been reliable for me.

I'll close this PR to free up the resources and use this a an opportunity try to find out how hard it is to self-host a flatpak repo. Thank you all for taking your time to review this!

**Pull request closed without merging**